go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -782,14 +810,19 @@ func (root *rootValue) PutTable(ctx context.Context, tName TableName, table *Tab
 		return nil, err
 	}
 
-	return root.putTable(ctx, tName, tableRef)
+	schHash, err := table.GetSchemaHash(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return root.putTable(ctx, tName, tableRef, schHash)
 }
 
 func RefFromNomsTable(ctx context.Context, table *Table) (types.Ref, error) {
 	return durable.RefFromNomsTable(ctx, table.table)
 }
 
-func (root *rootValue) putTable(ctx context.Context, tName TableName, ref types.Ref) (RootValue, error) {
+func (root *rootValue) putTable(ctx context.Context, tName TableName, ref types.Ref, schHash hash.Hash) (RootValue, error) {
```

Comment:
Would get rid of this unused caching logic and the new param. It has minimal benefit like you note, and it's a leaky abstraction.

If we want to cache this value across table updates, the coordinating logic should live somewhere else, probably the session.

Reply:
Not sure I follow. (1) It's used, (2) I added because I thought it's beneficial, getting the schema hash via a table lookup isn't cheap, (3) the comment is about a more aggressive degree of sharing that still externally consistent, but less internally consistent, and (4) putting the check here seems like the least leaky way to do this, the root value is immutable and its lifecycle controls the cache consistency